src/Hyper/Node/Session/InMemory.purs

```diff
@@ -31,12 +31,12 @@ instance sessionStoreInMemorySessionStore :: ( Monad m
   put (InMemorySessionStore var) id session = do
     liftAff do
       log ("Saving session: " <> unwrap id)
-      Map.insert id session <$> readVar var >>= flip putVar var
+      Map.insert id session <$> takeVar var >>= flip putVar var
 
   delete (InMemorySessionStore var) id = do
     liftAff do
       log ("Deleting session: " <> unwrap id)
-      Map.delete id <$> readVar var >>= flip putVar var
+      Map.delete id <$> takeVar var >>= flip putVar var
```

Comment:
I'd recommend using `tryTakeVar` here instead, as it will take it only if there is something to take. In general, I'd recommend just using a `Ref` though unless you really need the async coordination.

Reply:
Sorry, ignore that comment. Since this is just mutation, and on node, I really would recommend a `Ref` instead.

Reply:
Thank you for your advice. I just failed when I run session example. I use purescript-debug to find this bug. But your advice is more important for me to master `Aff`. I plan to deep into `MVar` in Haskell first. And then I will refactor my code, and push again. 

Thank you @natefaubion